Firestore/core/src/util/reader_context.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef FIRESTORE_CORE_SRC_UTIL_READER_CONTEXT_H_
```

Comment:
Nit: to reflect the class name correctly, the file name should be `read_context`.

To be honest, now that this class is detached from its original use, it seems like more of a generic `Context` -- there isn't really anything in the interface that makes it specifically a "read" context. No action necessary here, though -- feel free to keep the current name.

Reply:
I agree.